cornflow-dags/DAG/rostering/core/instance.py

```diff
@@ -338,18 +372,14 @@ def _hour_to_slot(self, hours) -> int:
         """Converts from a hours to slots"""
         return int(hours * 60 / self._get_slot_length())
 
-    def _get_minutes(self, s) -> float:
-        """Method to get the number of minutes from the start of the day given a slot"""
-        return self._get_starting_hour() * 60 + s * self._get_slot_length()
-
     def _get_employees_contracts(self) -> SuperDict[Tuple[int, int], int]:
         """
         Returns a SuperDict with the week and employee tuple as key
         and the contract id as value
         Contracts are supposed to start on Monday and end on Sunday
         For example: {(36, 1): 10, ...}
         """
-        default_date = get_date_string_from_ts(self._get_end_date())
+        default_date = get_date_string_from_ts(self.dates[-1])
```

Comment:
I know that dates is sorted, but I would prefer that instead of it being `self.dates[-1]` we keep a `self._get_end_date()` which in turns either it returns the maximum date in `self.dates` to make sure that even if something happens and it is not properly sorted that we get the last date from the horizon.